### ISSUE TITLE
RestService RegisterSchemaAsync missing schema reference support for Avro

### ISSUE THREAD

**Issue by jonhwong** [confluentinc/confluent-kafka-dotnet#1716]
Description
===========
RegisterSchemaAsync is not passing the `Schema` argument as is if the SchemaType is Avro and therefore does not send schema references. It is commented to be backward compatible but is not compatible with https://docs.confluent.io/platform/current/schema-registry/develop/api.html#subjects.

From branch 1.8.x
```
public async Task<int> RegisterSchemaAsync(string subject, Schema schema)
    => schema.SchemaType == SchemaType.Avro
        // In the avro case, just send the schema string to maintain backards compatibility.
        ? (await RequestAsync<SchemaId>($"subjects/{WebUtility.UrlEncode(subject)}/versions", HttpMethod.Post, new SchemaString(schema.SchemaString))
                .ConfigureAwait(continueOnCapturedContext: false)).Id
        : (await RequestAsync<SchemaId>($"subjects/{WebUtility.UrlEncode(subject)}/versions", HttpMethod.Post, schema)
                        .ConfigureAwait(continueOnCapturedContext: false)).Id;
```

How to reproduce
================
1 Call `RegisterSchemaAsync(string subject, Schema schema)` on `Confluent.SchemaRegistry.CachedSchemaRegistryClient` with
`Schema` argument:
```
{
  "schema": "[\"Sample.User.UserCreated\",\"Sample.User.UserEnabled\",\"Sample.User.UserDisabled\"\r\n]",
  "references": [
    {
      "name": "Sample.User.UserCreated",
      "subject": "Sample.User.UserCreated",
      "version": 1
    },
    {
      "name": "Sample.User.UserEnabled",
      "subject": "Sample.User.UserEnabled",
      "version": 1
    },
    {
      "name": "Sample.User.UserDisabled",
      "subject": "Sample.User.UserDisabled",
      "version": 1
    }
  ],
  "schemaType": "AVRO"
}
```
2 Receive Exception error code 42201

**Comment by mhowlett** [comment]
Not supported in .NET. AFAIK, this won't be easy to implement.

**Comment by rayokota** [comment]
I believe this is fixed by #2226
